packages/view/backend/sirius-components-view-emf/src/main/java/org/eclipse/sirius/components/view/emf/table/ViewRowContextMenuEntryExecutor.java

```diff
@@ -0,0 +1,140 @@
+
+    private final IViewAQLInterpreterFactory aqlInterpreterFactory;
+
+    private final IFeedbackMessageService feedbackMessageService;
+
+    private final IEditService editService;
+
+    private final IObjectSearchService objectSearchService;
+
+    private final Function<EObject, UUID> idProvider = (eObject) -> UUID.nameUUIDFromBytes(EcoreUtil.getURI(eObject).toString().getBytes());
+
+    public ViewRowContextMenuEntryExecutor(ViewRepresentationDescriptionPredicate viewRepresentationDescriptionPredicate,
+            IViewRepresentationDescriptionSearchService viewRepresentationDescriptionSearchService, IViewAQLInterpreterFactory aqlInterpreterFactory, IFeedbackMessageService feedbackMessageService,
+            IEditService editService, IObjectSearchService objectSearchService) {
+        this.viewRepresentationDescriptionPredicate = Objects.requireNonNull(viewRepresentationDescriptionPredicate);
+        this.viewRepresentationDescriptionSearchService = Objects.requireNonNull(viewRepresentationDescriptionSearchService);
+        this.aqlInterpreterFactory = Objects.requireNonNull(aqlInterpreterFactory);
+        this.feedbackMessageService = Objects.requireNonNull(feedbackMessageService);
+        this.editService = Objects.requireNonNull(editService);
+        this.objectSearchService = Objects.requireNonNull(objectSearchService);
+    }
+
+    @Override
+    public boolean canExecute(TableDescription tableDescription, String tableId, String rowId, String rowMenuContextEntryId) {
+        return this.viewRepresentationDescriptionPredicate.test(tableDescription);
+    }
+
+    @Override
+    public IStatus execute(IEditingContext editingContext, TableDescription tableDescription, Table table, Line row, String rowMenuContextEntryId) {
+        var optionalTableDescription = this.viewRepresentationDescriptionSearchService
+                .findById(editingContext, tableDescription.getId())
+                .filter(org.eclipse.sirius.components.view.table.TableDescription.class::isInstance)
+                .map(org.eclipse.sirius.components.view.table.TableDescription.class::cast);
+        if (optionalTableDescription.isPresent()) {
+            var viewTableDescription = optionalTableDescription.get();
+
+            AQLInterpreter interpreter = this.aqlInterpreterFactory.createInterpreter(editingContext, (View) viewTableDescription.eContainer());
+
+            VariableManager variableManager = new VariableManager();
+            variableManager.put(IEditingContext.EDITING_CONTEXT, editingContext);
+            variableManager.put(TableDescription.TABLE, table);
+            variableManager.put(LineDescription.SELECTED_ROW, row);
+            variableManager.put(LineDescription.ID, row.getId());
+            var optionalSemanticObject = this.objectSearchService.getObject(editingContext, row.getTargetObjectId());
+            if (optionalSemanticObject.isPresent()) {
```

Comment:
Why does the operation cannot be performed if the variable self is not available?